truss/cli/train/deploy_checkpoints/deploy_checkpoints.py

```
@@ -123,82 +119,60 @@ def _ensure_deployment_name(
-            truss_deploy_config.secrets[value.name] = "set token in baseten workspace"
-            start_command_envvars = f"{key}=$(cat /secrets/{value.name})"
-
-    checkpoint_parts = []
-    for truss_checkpoint in truss_deploy_config.training_checkpoints.checkpoints:  # type: ignore
-        ckpt_path = Path(
-            truss_deploy_config.training_checkpoints.download_folder,  # type: ignore
-            truss_checkpoint.id,
-        )
-        checkpoint_parts.append(f"{truss_checkpoint.name}={ckpt_path}")
-    checkpoint_str = " ".join(checkpoint_parts)
-    max_lora_rank = max(
-        [
-            checkpoint.lora_rank or DEFAULT_LORA_RANK
-            for checkpoint in checkpoint_deploy.checkpoint_details.checkpoints
-        ]
-    )
-    accelerator = checkpoint_deploy.compute.accelerator
-    if accelerator:
-        specify_tensor_parallelism = f" --tensor-parallel-size {accelerator.count}"
+
+def _render_truss_config_for_checkpoint_deployment(
+    checkpoint_deploy: DeployCheckpointsConfigComplete,
+) -> truss_config.TrussConfig:
+    """
+    Render truss config for checkpoint deployment.
+    Currently supports LoRA checkpoints via vLLM, but can be extended for other formats.
+    """
+    # Delegate to specific rendering function based on model weight format
+    if checkpoint_deploy.model_weight_format == truss_config.ModelWeightsFormat.LORA:
+        return render_vllm_lora_truss_config(checkpoint_deploy)
     else:
-        specify_tensor_parallelism = ""
-
-    start_command_args = {
-        "base_model_id": checkpoint_deploy.checkpoint_details.base_model_id,
-        "lora_modules": checkpoint_str,
-        "envvars": start_command_envvars,
-        "max_lora_rank": max_lora_rank,
-        "specify_tensor_parallelism": specify_tensor_parallelism,
-    }
-    truss_deploy_config.docker_server.start_command = VLLM_LORA_START_COMMAND.render(
-        **start_command_args
-    )
-    return truss_deploy_config
+        raise ValueError(
```

Should we throw an error here? Can we generate create a truss config instead? Or instruct the user to run with `--truss-config` so that it's intentional?

---

It's not actually possible to enter that branch. If you run `truss train deploy_checkpoints`, you're required to select either LoRA or, in the future, one of the valid `model_weight_format` options. If you try to run a command like:

```bash
truss train deploy_checkpoints --config="deploy_config.py" --remote=baseten
```

but pass an invalid `model_weight_format`, PyDantic will immediately raise a validation error.

I could delete that code path entirely, but I’ve kept it around as a backup for now.

---

got it ok. We can follow up with something more arbitrary